src/model/documentselection.js

```diff
@@ -707,9 +707,15 @@ function clearAttributesStoredInElement( changes, batch ) {
 		return;
 	}
 
-	const storedAttributes = Array.from( changeParent.getAttributeKeys() ).filter( key => key.startsWith( storePrefix ) );
+	const anyStoredAttributes = Array.from( changeParent.getAttributeKeys() ).some( key => key.startsWith( storePrefix ) );
```

Comment:
I think that we should make an agreement whether we are fine with `enqueueChanges` calls that do nothing. If we are fine with them, then we should not make conditional statements before `enqueueChanges`. The drawback is that we may fire more `changesDone` than needed, though.

If we are fine with empty `enqueueChanges` calls, we can remove that `if`.